### ISSUE TITLE
KeyError: u'UnboundedIndexRangeEncode'

### ISSUE THREAD

**Issue by chansongoal** [tensorflow/compression#22]
Hi, 
I try to compress an image by running the command at 'example' folder: "python tfci.py compress mbt2018-mean-mse-1 test.png". But it failed. I did not modify anything to the original codes. Do you have any idea about it? 

Environment:
python 2.7
tensorflow1.13-gpu

Error Message:
Traceback (most recent call last):
  File "tfci.py", line 326, in <module>
    app.run(main, flags_parser=parse_args)
  File "/home/gcs/App/anaconda3/envs/tensorflow1.13/lib/python2.7/site-packages/absl/app.py", line 300, in run
    _run_main(main, args)
  File "/home/gcs/App/anaconda3/envs/tensorflow1.13/lib/python2.7/site-packages/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "tfci.py", line 318, in main
    args.target_bpp, args.bpp_strict)
  File "tfci.py", line 154, in compress
    bitstring = compress_image(model, input_image)
  File "tfci.py", line 110, in compress_image
    signature_defs = import_metagraph(model)
  File "tfci.py", line 88, in import_metagraph
    tf.train.import_meta_graph(metagraph)
  File "/home/gcs/App/anaconda3/envs/tensorflow1.13/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 1435, in import_meta_graph
    meta_graph_or_file, clear_devices, import_scope, **kwargs)[0]
  File "/home/gcs/App/anaconda3/envs/tensorflow1.13/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 1457, in _import_meta_graph_with_return_elements
    **kwargs))
  File "/home/gcs/App/anaconda3/envs/tensorflow1.13/lib/python2.7/site-packages/tensorflow/python/framework/meta_graph.py", line 806, in import_scoped_meta_graph_with_return_elements
    return_elements=return_elements)
  File "/home/gcs/App/anaconda3/envs/tensorflow1.13/lib/python2.7/site-packages/tensorflow/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/home/gcs/App/anaconda3/envs/tensorflow1.13/lib/python2.7/site-packages/tensorflow/python/framework/importer.py", line 399, in import_graph_def
    _RemoveDefaultAttrs(op_dict, producer_op_list, graph_def)
  File "/home/gcs/App/anaconda3/envs/tensorflow1.13/lib/python2.7/site-packages/tensorflow/python/framework/importer.py", line 159, in _RemoveDefaultAttrs
    op_def = op_dict[node.op]
KeyError: u'UnboundedIndexRangeEncode'

**Comment by zzktb** [comment]
Hi chansongoal, did you get this problem resolved? I encountered the same problem while trying to load the MetaGraph file (bmshj2018-hyperprior-msssim-8). The code I'm using is attached. Any help would be highly appreciated.
```
def load_metagraph(file):
    """
    Load metagraph file.
    :param file:
    :return: MetaGraphDef protocol buffer
    """
    with tf.io.gfile.GFile(file, 'rb') as f:
        string = f.read()

    meta_graph = tf.MetaGraphDef()
    meta_graph.ParseFromString(string)
    tf.train.import_meta_graph(meta_graph)
    return meta_graph
```

**Comment by chansongoal** [comment]
Did you install the tensorflow_compression in your local python environment？I have solved this problem through the command “pip install tensorflow_compression”. Another tip for this project：run it in docker environment.

**Comment by chansongoal** [comment]
Did you install the tensorflow_compression in your local python environment？I have solved this problem through the command “pip install tensorflow_compression”. Another tip for this project：run it in docker environment.

On Aug 15, 2019, at 6:04 AM, Zhaobin <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi chansongoal, did you get this problem resolved? I encountered the same problem while trying to load the MetaGraph file (bmshj2018-hyperprior-msssim-8). The code I'm using is attached. Any help would be highly appreciated.

def load_metagraph(file):
    """
    Load metagraph file.
    :param file:
    :return: MetaGraphDef protocol buffer
    """
    with tf.io.gfile.GFile(file, 'rb') as f:
        string = f.read()

    meta_graph = tf.MetaGraphDef()
    meta_graph.ParseFromString(string)
    tf.train.import_meta_graph(meta_graph)
    return meta_graph
